benchexec/tools/map2check.py

```diff
@@ -27,20 +28,48 @@ class Tool(benchexec.tools.template.BaseTool):
     This class serves as tool adaptor for Map2Check (https://github.com/hbgit/Map2Check)
     """
 
-    REQUIRED_PATHS = [
+    REQUIRED_PATHS_6 = [
                   "__init__.py",
                   "map2check.py",
                   "map2check-wrapper.sh",
                   "modules"
                   ]
 
+    REQUIRED_PATHS_7_1 = [
+                  "__init__.py",
+                  "map2check",
+                  "map2check-wrapper.py",
+                  "modules"
+                  ]
+
     def executable(self):
         #Relative path to map2check wrapper
-        return Util.find_executable('map2check-wrapper.sh')
+        if self._get_version() == 6:
+            return Util.find_executable('map2check-wrapper.sh')
+        elif self._get_version() > 6:
+            return Util.find_executable('map2check-wrapper.py')
+
 
     def program_files(self, executable):
         executableDir = os.path.dirname(executable)
-        return [executableDir]
+
+        if self._get_version() == 6:
+            paths = REQUIRED_PATHS_6
+        elif self._get_version() > 6:
+            paths = REQUIRED_PATHS_7_1
+
+        return [executableDir] + paths
```

Comment:
I just happened to see this now: Aren`t the files from `REQUIRED_PATHS` inside your `executableDir`? If yes, this is redundant. In the old code it seems that `REQUIRED_PATHS` was completely unused. I would prefer to remove `executableDir` from the result of `program_files` to improve this.